linker/shared/src/main/scala/org/scalajs/linker/standard/IRFileImpl.scala

```diff
@@ -10,62 +10,27 @@
  * additional information regarding copyright ownership.
  */
 
-package org.scalajs.linker.irio
+package org.scalajs.linker.standard
 
 import scala.concurrent._
 
 import java.io.IOException
+import java.nio.ByteBuffer
 
 import org.scalajs.ir
 
-/** A virtual file containing Scala.js IR.
- *
- *  The main difference compared to using individual files
- *  (that are extracted beforehand) is that the fileset can be versioned at a
- *  higher level: the container needs to change its version when any of the
- *  files change. Therefore, the entire extraction process can be cached.
- */
-trait ScalaJSIRContainer {
-  /** Abstract path of the file.
-   *
-   *  The path of the file is used for lookup and caching (together with the
-   *  version).
-   */
-  val path: String
-
-  /** An optional implementation-dependent "version" token.
-   *
-   *  If non-empty, a different version must be returned when the content
-   *  changes. It should be equal if the content has not changed, but it is
-   *  not mandatory.
-   *  Such a token can be used by caches: the file need not be read and
-   *  processed again if its version has not changed.
-   */
-  val version: Option[String]
-
-  /** All the `*.sjsir` files in this container.
-   *
-   *  It is up to the implementation whether these files are read lazily or not.
-   */
-  def sjsirFiles(implicit ec: ExecutionContext): Future[List[VirtualScalaJSIRFile]]
-
-  override def toString(): String = {
-    val className = getClass.getName
-    val shortClassName = className.substring(className.lastIndexOf('.') + 1)
-    shortClassName + "(" + path + ")"
-  }
-}
+import org.scalajs.linker.IRFile
 
 /** A virtual Scala.js IR file.
  *  It contains the class info and the IR tree.
  */
-trait VirtualScalaJSIRFile {
+abstract class IRFileImpl (
```

Comment:
No space before `(` here. We only use a space if there is a visibility modifier for the constructor.